src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/UnprotectedAttributeTests.cs

```diff
@@ -268,6 +270,43 @@ private static void VerifyUnprotectedAttributes_Heterogenous3(byte[] encodedMess
         }
 
         [Fact]
+        public static void TestUnprotectedAttributes1_EmptySet()
+        {
+            // This tests the behavior of unprotected attribute extraction when one of the attribute sequences declares an
+            // attribute type, but the contained SET OF AttributeValue is empty.
+            // Attribute ::= SEQUENCE {
+            //      attrType OBJECT IDENTIFIER,
+            //      attrValues SET OF AttributeValue }
+            // The encoded message was built in ASN.1 editor and tested in framework.It contains an enveloped message
+            // version 2 with a key transport recipient, the enveloped message contains data encrypted with 3DES.
+            // The attributes set is built as
+            // {
+            //      { attrType: document description, attrValues: {} },
+            //      { attrType: document name, attrValue: { } },
+            // }
+            // The important part of this test is that there are 0 attributes of a type that is declared within the encoded message.
+            // On framework, this would return 2 as explained on issue 9252, that is it woulds create a CryptographicObjectCollection
+            // with two CryptographicObject, the first one holding a list of document description with the two values, the second one
+            // holding an empty list of document name. This makes ecms.UnprottectedAttributes.Count be 2. Core doesn't create the second
+            // list and as such ecms.UnprottectedAttributes.Count returns 1 as documented here.
```

Comment:
* You have unprotected typoed a couple of times.
* CryptographicObject => CryptographicAttributeObject (I think)
* I thought each AttributeValue entry resulted in a new CryptographicAttributeObject, so if it's 2+0 I'd expect 3 objects for Framework, 2 for core.  Am I mistaken?